gzip.go

```diff
@@ -258,6 +274,45 @@ func NewGzipLevelAndMinSize(level, minSize int) (func(http.Handler) http.Handler
 	}, nil
 }
 
+// Used for functional configuration.
+type GzipConfig struct {
```

Comment:
Does this need to be public? Also, to reduce stutter, can we just call this `config`?